src/promptflow/promptflow/executor/_dag_manager.py

```diff
@@ -110,11 +110,10 @@ def _is_node_bypassable(self, node: Node) -> bool:
             return True
 
         # Bypass node if the activate condition is not met
-        if node.activate and (
-            self._is_node_dependency_bypassed(node.activate.condition)
-            or not self._is_condition_met(node.activate.condition, node.activate.condition_value)
-        ):
-            return True
+        if node.activate:
+            if self._is_node_dependency_bypassed(node.activate.condition):
+                return True
```

Comment:
Not sure I understand this correctly, does this mean the node would still be bypassed if its dependency is bypassed?

Reply:
Here the logic "is_node_dependency_bypassed" is checking whether the activate condition is referencing another node but the node is bypassed. It is still about activate condition, not the input dependencies.

@PeiwenGaoMS Could add more clear comments here.

Reply:
Added.

Reply:
I see. It's bad naming, I'm not sure any newcomer would understand the logic in your way.

Would it be better to rename to something like "_is_referenced_node_in_condition_bypassed"?

Reply:
`_is_referenced_node_in_condition_bypassed` is inappropriate because this function is not only used to determine whether the condition has been skipped. For a node, its dependencies are not only inputs, but also the node referred in skip config and activate config.

Reply:
Will app some comments to this function.

Reply:
Oh I see what you mean here, so the "dependency" in the function name is not for the "node" itself, but for the input parameter "node.activate.condition"
That's what confused me.
Maybe something like "_is_reference_node_bypassed" would be more friendly to newcomer.
Not blocker though.

Reply:
To elaborate a little more of my confusion:
As you said "For a node, its dependencies are not only inputs, but also the node referred in skip config and activate config", and if I understand the function as "whether _is_node_dependency_bypassed for a node", then I'm confused, since we clearly deal with input referenced and condition referenced node differently.